pytorch_lightning/callbacks/predictions.py

```diff
@@ -0,0 +1,90 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, List
+
+from pytorch_lightning.callbacks.base import Callback
+from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+
+class BasePredictionWriter(Callback):
```

Comment:
naming: we should either always postfix with `Base` or prefix with `Base`. given there's a `ProgressBarBase` this could be named `PredictionWriterBase`

Reply:
You are right ! Let's follow current PL convention.

Reply:
I wouldn't say we have a strict convention, see the occurrences:

Prefix:
`BasePruningMethod` (from PyTorch)
`BaseException` (from Python)
`BaseFinetuning`
`BaseProfiler`

Suffix:
`_LightningModuleWrapperBase`
`LightningLoggerBase`
`ProgressBarBase`

Reply:
let's pick one naming convention and stick with it to avoid issues like https://github.com/PyTorchLightning/pytorch-lightning/issues/6094